mono/mini/mini-llvm.c

```diff
@@ -9757,10 +9891,33 @@ mono_llvm_emit_aot_module (const char *filename, const char *cu_name)
 				/* The types might not match because the caller might pass an rgctx */
 				if (lmethod && LLVMTypeOf (callee) == LLVMTypeOf (lmethod)) {
 					mono_llvm_replace_uses_of (callee, lmethod);
+
+					g_hash_table_insert (directly_called, lmethod, ji->data.method);
 					mono_aot_mark_unused_llvm_plt_entry (ji);
 				}
 			}
 		}
+
+		// After direct calls are all computed, specialize directly called
```

Comment:
So there's an analysis pass that computes `directly_called`. And starting from this comment, this is kind of a global optimization pass over the current module.
Might be worth pulling it out into a separate function so that it's clear that (1) it's optional, (2) where to add another such optimization. 

It'd be nice also to document what kinds of LLVM IR manipulations would invalidate `directly_called` in case we do more global passes.